backend/service/chaos/experimentation/terminator/terminator.go

```diff
@@ -0,0 +1,156 @@
+package terminator
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+
+	experimentationv1 "github.com/lyft/clutch/backend/api/chaos/experimentation/v1"
+	"github.com/lyft/clutch/backend/service/chaos/experimentation/experimentstore"
+)
+
+type TerminationCriteria interface {
+	ShouldTerminate(experimentStarted time.Time, config interface{}) error
```

Comment:
Idea: Maybe we could allow `ShouldTerminate` method to return a termination reason that we would then record in a database. This way we could make these messages more meaningful and context-aware. Not a must tho.

Reply:
Yeah let me clean this up a bit. Right now it returns an `error` that we record, but maybe it would be clearer with (string, error) and not terminate on errors